src/main/scala/viper/gobra/translator/encodings/slices/SliceEncoding.scala

```diff
@@ -47,6 +47,28 @@ class SliceEncoding(arrayEmb : SharedArrayEmbedding) extends LeafTypeEncoding {
       case Shared => vpr.Ref
     }
   }
+  /**
+    * Encodes assertions.
+    * [acc(m: []T, perm)] -> [forall i int :: 0 <= i && i < len(m) ==> acc(&m[i], perm)]
+    */
+  override def assertion(ctx: Context): in.Assertion ==> CodeWriter[vpr.Exp] = {
+    default(super.assertion(ctx)) {
+      case n@ in.Access(in.Accessible.ExprAccess(exp :: ctx.Slice(elem)), perm) =>
+        val iterVar = in.BoundVar(Names.freshName, in.IntT(Addressability.Exclusive))(n.info)
+        val underlyingType = in.SliceT(elem, Addressability.exprInAcc)
+        val quantifiedAssert = in.SepForall(
+          vars = Vector(iterVar),
+          triggers = Vector(in.Trigger(Vector(in.IndexedExp(exp, iterVar, underlyingType)(n.info)))(n.info)),
```

Comment:
should the trigger be `m[i]` or `&m[i]`. To my understanding, `m[i]` is a heap trigger and in theory is better. However, my trust in heap trigger is not that big. You have experience with this in practice; Which of the two does work better for you?